### ISSUE TITLE
Enabling KSP2 causes the build to fail

### ISSUE THREAD

**Issue by nakamura-to** [komapper/komapper#1318]
### Describe the bug

Using KSP 2.0.0-1.0.23 and enabling the `ksp.useKSP2` flag causes the build to fail.

### Expected behavior

Build succeeds without errors.

### Actual behavior

The error message "@KomapperEntity cannot be applied to this element" is displayed.

### Steps to reproduce the problem

- Add `ksp.useKSP2=true` to the gradle.properties file.
- Create an entity class like the following: 
    ```kotlin
    @KomapperEntity
    data class Employee(
        @KomapperId @KomapperAutoIncrement
        val id: Int = 0,
        val name: String,
        @KomapperVersion
        val version: Int = 0,
        @KomapperCreatedAt
        val createdAt: LocalDateTime = LocalDateTime.MIN,
        @KomapperUpdatedAt
        val updatedAt: LocalDateTime = LocalDateTime.MIN,
    )
    ```
- Run the build.

### Versions

- Komapper: 2.0.0
- Kotlin: 2.0.0
- KSP: 2.0.0-1.0.23 
- JVM: 11

**Comment by nakamura-to** [comment]
The issue is likely related to the bug reported at https://github.com/google/ksp/issues/1996.
Until a new version of KSP is released, please use one of the following workarounds:

1. Turn off the `ksp.useKSP2` flag.
2. Use KSP version 2.0.0-1.0.22.

**Comment by nakamura-to** [comment]
It seems that this issue has been resolved in KSP v2.0.0-1.0.24.
